src/Dianoga/Png/PngQuantOptimizer.cs

```diff
@@ -24,11 +26,17 @@ public override IOptimizerResult Optimize(MediaStream stream)
 
 			using (var bitmap = new Bitmap(stream.Stream))
 			{
+				var bitDepth = Image.GetPixelFormatSize(bitmap.PixelFormat);
+				if (bitDepth != 32)
+				{
+					return OptimizerFailureResult("the image you are attempting to quantize does not contain a 32 bit ARGB palette. This image has a bit depth of {0} with {1} colors".FormatWith(bitDepth, bitmap.Palette.Entries.Length));
+				}
```

Comment:
nQuant is unable to optimize images unless it is 32 bit ARGB format. Handle this gracefully rather than letting the library throw an exception.